EricGray/homework.js

```diff
@@ -0,0 +1,386 @@
+//Part 1: A Cards Toolkit!>
+
+
+
+var suitARR = [" Hearts", " Diamonds", " Spades", " Clubs"];
+var rankARR = ["Ace ", "Two ", "Three ", "Four ", "Five ", "Six ", "Seven ", 
+"Eight ", "Nine ", "Ten ", "Jack ", "Queen ", "King "];
+
+function checkValidID(id) {
+	if (id < 0 || id > 51 || id%1 !== 0) {
+		alert("Sorry, your number is out of range. Only whole numbers 0-51, please!)");
+		return NaN;
+			} else if (typeof id === "string" || typeof id === "boolean"
+			 || typeof id === "undefined") {
+		alert("Sorry, only whole numbers 0-51, please!");
+		return NaN;
+	} else {
+		return id;
+	} 
+;}
+
```

Comment:
Looking good, but instead of writing validation tests again and again at the top of your functions you should use the single checkValidID() test you wrote at the top of the page. So for an example in your rank function you would right something like this:
```
    rank: function(id) {
        if (return this.isValid(id)) 
             return Math.floor(id/4)+1;
    }
```